src/net.cpp

```diff
@@ -1796,10 +1796,10 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
 
     CNode* pnode = new CNode(id,
                              std::move(sock),
-                             addr,
+                             CAddress{addr, NODE_NONE},
                              CalculateKeyedNetGroup(addr),
                              nonce,
-                             addr_bind,
+                             CAddress{addr_bind, NODE_NONE},
```

Comment:
`CNode::addr` needs to be `CAddress` because at least `PeerManagerImpl::PushNodeVersion()` is using `addr.nServices`.

`CNode::addrBind` indeed does not need to be `CAddress`. I changed it to `CService`.